### PR TITLE
54382 Rename Theme|Plugin Editor to Theme|Plugin File Editor

### PR DESCRIPTION
**Pull request by audrasjb**
Trac ticket: https://core.trac.wordpress.org/ticket/54382

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by hellofromtonya** [comment]
@audrasjb can you update this PR to fix the merge conflicts?

**Review by youknowriad — COMMENTED** [review]
I think renaming the menu entries for plugin and theme file editors for consistency makes sense to me.

But I stand with my reasoning that we shouldn't keep the the theme file editor under Appearance. All text can already be edited in the site editor with FSE themes, there's no need for these "tool" unless you're a developer or you want to break your site.

**Comment by audrasjb** [comment]
Conflicts resolved. The failing e2e test looks unrelated.

File: src/wp-admin/menu.php [hunk 1]
```diff
@@ -241,23 +241,23 @@
...
  * @since 3.0.0
- * @since 5.9.0 'Theme Editor' link has moved to the Tools menu when a block theme is active.
+ * @since 5.9.0 'Theme Editor' link was renamed to 'Theme File Editor'.
  */
 function _add_themes_utility_last() {
 	add_submenu_page(
-		wp_is_block_theme() ? 'tools.php' : 'themes.php',
-		__( 'Theme Editor' ),
-		__( 'Theme Editor' ),
+		wp_is_block_template_theme() ? 'tools.php' : 'themes.php',
```
Comment (youknowriad) [inline src/wp-admin/menu.php]: I think the previous name here is the right one.

Comment (audrasjb) [reply src/wp-admin/menu.php]: thanks!

**Review by youknowriad — CHANGES_REQUESTED** [review]
I think we should not move it back to appearance, having two ways to edit themes under appearance feels couter productive to me.

cc @jasmussen @jameskoster @mtias

**Comment by audrasjb** [comment]
> I think we should not move it back to appearance, having two ways to edit themes under appearance feels couter productive to me.
> cc @jasmussen @jameskoster @mtias

Please note that it's also discussed in the related Trac ticket: https://core.trac.wordpress.org/ticket/54382

I proposed a compromise for 5.9 but it appears there is more committers voices supporting moving it back under Appearance menu.

**Comment by jasmussen** [comment]
To echo Riad, and to restate my comments from the trac ticket, I would be concerned with having two theme editing tools under appearance, one of them opening with a warning modal and risking changes being overwritten by theme updates.

**Comment by jameskoster** [comment]
Situations in which you'd need to use the file editor whilst a block theme is active seem almost non-existent. Especially because editing the source file won't have any effect if you've also edited the corresponding entity in the Site Editor already. This could make for some _very_ confusing experiences. 

At the very least I think the file editor should be de-emphasised when a block theme is active.

**Comment by hellofromtonya** [comment]
Folks thanks for voicing opinions. However, [see @johnbillion's comment](https://core.trac.wordpress.org/ticket/54382#comment:15) as to why this PR restores the Theme File Editor:

>Moving existing menu items causes a burden for end users. The existing Theme Editor is not a feature we can retire just yet. Take a look at how real users customise their site and you'll see usage of the Theme Editor many times higher than us contributors would like to imagine. Moving this menu item creates an inconsistency when switching between FSE-enabled themes and existing themes, and it will place burden on end users who expect it to be where it always is and on support volunteers in the forums.

**Comment by hellofromtonya** [comment]
Looks like this PR is blocked for Beta 3. Needs to be resolved by Beta 4.

**Comment by jameskoster** [comment]
I don't think anyone is suggesting to retire the feature (although I would be in favor of that personally). This is more an exercise in adjusting its prominence to suit an evolving site editing landscape. 

To me this feels like an extension of the decision to conditionally hide the Customizer. If we apply the principles that lead to that decision here, then it seems sensible to be consistent, and do something similar with the theme file editor.

I totally appreciate the desire to minimise any disruption to current user flows and honour existing documentation. But if the trade-off is a confusing in-product user experience then it seems like a direction worth questioning.

**Comment by 0aveRyan** [comment]
I hear the argument for making this change (edit: to remove or relocate the legacy file editor link), but it should have been made months ago to make this release. The Customizer decision was made with ample time for the ecosystem to prepare. Last minute changes without ample time for input and preparation are confidence damaging. I think this change should be made iteratively and targeted for 6.0.

**Comment by hellofromtonya** [comment]
> To echo Riad, and to restate my comments from the trac ticket, I would be concerned with having two theme editing tools under appearance, one of them opening with a warning modal and risking changes being overwritten by theme updates.

That same argument though can made for Customizer and Theme Editor menu items that existed in the Appearance menu before 5.9. 

>To me this feels like an extension of the decision to conditionally hide the Customizer. If we apply the principles that lead to that decision here, then it seems sensible to be consistent, and do something similar with the theme file editor.

The Site Editor is _replacing_ (and hiding) the Customizer for block themes.

Users have had 2 menu items under Appearance to customize their site via (a) selective/interactive interface or (b) via directly editing files within a file editing interface.
* Pre-5.9: Customize menu was (a) and the Theme Editor menu was for (b).
* 5.9: Site Editor is (a) and the Theme File Editor is still for (b).

These 2 menu items are different. The way each presents is very different. It's clear that Customizer or Site Editor is very different from the file editor interface with `Theme File Editor`. Personally, I don't see a workflow issue or why keeping the file editor in place would cause confusion or concern, whereas moving it could cause a burden on extenders and users as information has existed for years and is present across the web of what the Theme Editor is and what it does.

If there's a need to relocate these file editors (i.e. the theme and plugin versions), 6.0 would be a good target. That gives more time to collect user feedback and analyze UX workflows to determine if a problem exists or not with the file editor existing in the same Appearance menu space.

**Review by hellofromtonya — DISMISSED** [review]
Marking for commit.

**Comment by jameskoster** [comment]
I disagree, and see the Site Editor as a visual evolution of the file editor as much as a replacement for the Customizer. Since it is able to interpret block theme files, I can think of very few situations in which you would want to edit the source files in wp-admin. Consequently it's a feature we should consider de-emphasising. Of course it's quite possible I'm missing something so would be keen to understand common use cases for the file editor.

As for confusing scenarios, here's one: You edit your Index template in the Site Editor, (creating a custom Index template in your database behind the scenes). You then go and edit the same Index template in the file editor, but the contents are different (because the site editor does not overwrite the original file) and your changes do not manifest on the frontend (because the template in the db is prioritised over the theme file). Without an understanding of the underlying technicals it's not clear at all why these editors behave differently.

**Comment by hellofromtonya** [comment]
>As for confusing scenarios, here's one: You edit your Index template in the Site Editor, (creating a custom Index template in your database behind the scenes). You then go and edit the same Index template in the file editor, but the contents are different (because the site editor does not overwrite the original file) and your changes do not manifest on the frontend (because the template in the db is prioritised over the theme file). Without an understanding of the underlying technicals it's not clear at all why these editors behave differently.

@jameskoster thank you for raising this scenario. You're right.

A thought came to my mind this morning. The evolution is towards no-code. FSE with the Site Editor is part of that evolution. There's a solid argument to made that file/code editor interfaces do not fit within the next generation no-code workflows. These file/code editor interfaces are rather tools, and thus are better aligned under `Tools`.

Given the valid user confusion scenarios James raised combined with the next step forward towards no-code workflows in 5.9, I can see and now do support moving this file/code editor interfaces to Tools.

Next questions:
To avoid confusion, should the menu item be changed to `Theme File Editor` to more clearly denote what it does? And should the `Plugin File Editor` menu item also be relocated under Tools?

I'm thinking yes to both of these questions.

File: src/wp-admin/menu.php [hunk 2]
```diff
@@ -241,23 +241,23 @@
...
  * @since 3.0.0
- * @since 5.9.0 'Theme Editor' link has moved to the Tools menu when a block theme is active.
+ * @since 5.9.0 'Theme Editor' link was renamed to 'Theme File Editor'.
  */
 function _add_themes_utility_last() {
 	add_submenu_page(
-		wp_is_block_theme() ? 'tools.php' : 'themes.php',
-		__( 'Theme Editor' ),
-		__( 'Theme Editor' ),
+		'themes.php',
```
Comment (hellofromtonya) [inline src/wp-admin/menu.php]: @audrasjb See my comment/thinking here https://github.com/WordPress/wordpress-develop/pull/1948#issuecomment-998943854. I now agree that the Theme File Editor is a tool that belongs under "Tools".

```suggestion
		wp_is_block_theme() ? 'tools.php' : 'themes.php',
```

**Review by hellofromtonya — APPROVED** [review]
The phpcs issue is unrelated to this PR. Marking this for commit 👍

**Comment by hellofromtonya** [comment]
Folks, there was an open, collaborative discussion in Making WordPress slack core channel today. Attendees included @johnbillion, @costdev, @Clorith, @ipstenu, @audrasjb, @pbiron, and others. Consensus reached to:

* Keep the Theme Editor under Tools for block themes
* Move the Plugin Editor too for block themes (for consistency)
* Rename both to include "File" in their menu names

Retaining the change to relocate these file editors to Tools aligns with direction and consensus from the core editor team. Thus, this PR is now unblocked and ready for Beta 4.

**Comment by hellofromtonya** [comment]
Committed via changeset https://core.trac.wordpress.org/changeset/52406.

**Pull request closed without merging**